numba_typing/tests/test_type_annotations.py

```diff
@@ -0,0 +1,46 @@
+import unittest
+import type_annotations
+from typing import Union, Dict, List
+
+
+class TestTypeAnnotations(unittest.TestCase):
+
+    def test_get_func_annotations_exception(self):
+
+        def foo(a: int, b, c: str = "string"):
+            pass
+        self.assertRaises(SyntaxError,  type_annotations.get_func_annotations, foo)
+
+    def test_get_cls_annotations(self):
+        class MyClass(object):
+            x: int = 3
+            y: str = "string"
+
+            def __init__(self, x: str, y: int):
+                self.x = x
+                self.y = y
+
+        self.assertEqual(type_annotations.get_cls_annotations(MyClass), ({'x': int, 'y': str}, {}))
+
+    def test_get_func_annotations(self):
+        def func_one(a: int, b: Union[int, float], c: str):
+            pass
+        with self.subTest("annotations"):
+            self.assertEqual(type_annotations.get_func_annotations(func_one),
+                             ({'a': int, 'b': [int, float], 'c': str}, {}))
+
+        def func_two(a: int = 2, b: str = "string", c: List[int] = [1, 2, 3]):
+            pass
+        with self.subTest("annotations and all default values"):
+            self.assertEqual(type_annotations.get_func_annotations(func_two),
+                             ({'a': int, 'b': str, 'c': List[int]}, {'a': 2, 'b': 'string', 'c': [1, 2, 3]}))
+
+        def func_three(a: Dict[int, str], b: str = "string", c: int = 1):
+            pass
+        with self.subTest("annotations and not all default values"):
+            self.assertEqual(type_annotations.get_func_annotations(func_three),
+                             ({'a': Dict[int, str], 'b': str, 'c': int}, {'b': 'string', 'c': 1}))
```

Comment:
Expected something like that:
```suggestion
    def test_get_func_annotations(self):
        def func_one(a: int, b: Union[int, float], c: str):
            pass

        def func_two(a: int = 2, b: str = "string", c: List[int] = [1, 2, 3]):
            pass

        def func_three(a: Dict[int, str], b: str = "string", c: int = 1):
            pass

        expected_results = {
            func_one: ({'a': int, 'b': [int, float], 'c': str}, {}),
            func_two: ({'a': int, 'b': str, 'c': List[int]}, {'a': 2, 'b': 'string', 'c': [1, 2, 3]}),
            func_three: ({'a': Dict[int, str], 'b': str, 'c': int}, {'b': 'string', 'c': 1}),
        }
        for f, expected in expected_results.items():
            with self.subTest(func=f.__name__):
                self.assertEqual(type_annotations.get_func_annotations(f), expected)
```